### PR TITLE
Move loader functionality to Synchronizer - Closes #4783

### LINKED ISSUE #4783: Move loader to synchronizer
### Description
Loader functionality to fetch the transaction pool data should be moved to the `init` on synchronizer

### Motivation
Loader is currently doing the transactions/signatures fetch from other node on the startup.
Main purpose of the loader was to download information from other node.
Now, most of the feature is moved to the synchronizer, rest of the stuff in the loader should also move to the synchronizer and removed.

### Acceptance Criteria

- loader.js should be removed
- downloading the transactions and signatures should work as it did before
- `node/schema/definitions.js` should also defined in the synchronizer

### PR DESCRIPTION
**Pull request by ishantiw**
### What was the problem?
Loader functionality to be moved to Synchronizer.
This PR resolves #4783 

### How was it solved?

- Move `loadUnconfirmedTransactions` of `loader` to `synchronizer`
- Pass `transactionPool` and `channel` module to `synchronizer`'s constructor
- Rename `network` event `network:ready` -> `app:networkReady`
- Start `loadUnconfirmedTransactions` in `synchronizer`'s `init()` on `app:networkReady`
- Remove `loader.js` tests
- Add `loadUnconfirmedTransactions` tests to `synchronizer`
- Remove references to `loader` module

### How was it tested?

- Run all the tests
- Run `node framework/test/test_app`, it should run without any errors

### CODE REVIEW

File: framework/src/application/node/synchronizer/synchronizer.js [hunk 1]
```diff
@@ -15,26 +15,34 @@
...
 		);
 		this.mechanisms = mechanisms;
+		this.channel = channel;
 		this.logger = logger;
 		this.blocksModule = blocksModule;
 		this.processorModule = processorModule;
+		this.transactionPoolModule = transactionPoolModule;
 		this.storageModule = storageModule;
 		this.active = false;
+		this.loadTransactionsRetries = 5;
```
Comment (ManuGowda) [inline framework/src/application/node/synchronizer/synchronizer.js]: could we make this a constant?

File: framework/src/application/node/synchronizer/synchronizer.js [hunk 2]
```diff
@@ -141,6 +153,93 @@ class Synchronizer {
 
 		return undefined;
 	}
+
+	async loadUnconfirmedTransactions() {
+		for (const index of new Array(this.loadTransactionsRetries).keys()) {
```
Comment (ManuGowda) [inline framework/src/application/node/synchronizer/synchronizer.js]:
```
while (this.loadTransactionsRetries >= 0) {
this.loadTransactionsRetries --;
  try {
				await this._getUnconfirmedTransactionsFromNetwork();

				break;
			} catch (err) {
				if (err && index === this.loadTransactionsRetries - 1) {
					this.logger.error(
						{ err },
						`Failed to get transactions from network after ${this.loadTransactionsRetries} retries`,
					);
				}
			}
}
```

Comment (shuse2) [reply framework/src/application/node/synchronizer/synchronizer.js]: I think simple `for (let retry = 0; retry < this. loadTransactionsRetries; retry ++)` is a lot simpler

File: framework/src/application/node/synchronizer/synchronizer.js [hunk 3]
```diff
@@ -53,6 +61,10 @@ class Synchronizer {
 	}
 
 	async init() {
+		this.channel.subscribe('app:networkReady', async () => {
```
Comment (shuse2) [inline framework/src/application/node/synchronizer/synchronizer.js]: lets move this to `node.js`

File: framework/src/application/node/synchronizer/synchronizer.js [hunk 4]
```diff
@@ -15,26 +15,34 @@
 'use strict';
 
 const assert = require('assert');
+const { validator } = require('@liskhq/lisk-validator');
+const { Status: TransactionStatus } = require('@liskhq/lisk-transactions');
+const definitions = require('../schema/definitions');
```
Comment (shuse2) [inline framework/src/application/node/synchronizer/synchronizer.js]: can we move this definitions to `synchronizer/schema.js`?

File: framework/src/application/node/synchronizer/synchronizer.js (hunk 1 above)
Comment (ishantiw) [reply framework/src/application/node/synchronizer/synchronizer.js]: Do you mean that this constant should come from config?

**Pull request merged**